File: scrapers/extractedimage/japancoll.pl
{
	searchURL => 'http://www.japancollection.com/japanese-prints-search/allprices-allartists-allsubjects.php?y=1&ppp=100&pg=%s',
	searchStart => 1,
	searchEnd => 52,
	searchIncrement => 1,

	urlExtractor => scraper {
		process "//td/a[img]", 'urls[]' => '@href';
	},

	cleanURL => sub {
		my $url = shift;

		return $url =~ /pid=(\d+)/ ?
			"http://www.japancollection.com/japanese-prints-uview/print.php?pid=$1" :
			"";
	},

	extractPageID => qr/pid=(\d+)/,

	pageExtractor => scraper {
		process '//a[img][contains(@href, "jpg")]', source_image => '@href';
	},

	cleanPage => sub {
		my $item = shift;
		return $item;
	},

	extractImageFile => qr/([^\/]+)$/,

	cleanImageFile => sub {
		my $file = shift;
		$file =~ s/[: ]/_/g;
		return $file;
	}
}

File: scrapers/extractedimage/japancoll.js
exports.extract = {
	title: '//td[contains(.,"Title:")]/following-sibling::td',
	artist: '//td[contains(.,"Artist:")]/following-sibling::td',
	date: '//td[contains(.,"Date:")]/following-sibling::td',
	description: '//td[contains(.,"Series:")]/following-sibling::td',
	medium: '//td[contains(.,"Medium:")]/following-sibling::td',
	source_image: '//a[img][contains(@href, "jpg")]/@href'
};

exports.accept = data => {
	var pass = !/\S/.test(data.medium) || /Print/i.test(data.medium);
	delete data.medium;
	return pass;
};

exports.genURL = id => "http://www.japancollection.com/japanese-prints-uview/print.php?pid=" + id;
